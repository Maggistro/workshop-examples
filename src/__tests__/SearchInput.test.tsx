import React, { RefObject } from 'react';
import {render, mount} from 'enzyme';
import SearchInput from '../example/SearchInput';
import { SuggestionInput } from '../example/SuggestionInput';


const getDummyEvent = (value: string) => {
  return {currentTarget: {value: value}, target: {value: value}};
}

describe('Test for SearchInput', () => {
    it('should render the complete tree', () => {
        const searchInput = render(<SearchInput/>);
        expect(searchInput.html()).toMatchSnapshot();
    })

    it('should focus child components', () => {
        const searchInput = mount(<SearchInput/>)
        searchInput.find('input').at(0).simulate('change', getDummyEvent('test'));
        searchInput.find('input').at(1).simulate('change', getDummyEvent('test'));

        const ref0 = searchInput.find(SuggestionInput).at(0).props().innerRef as RefObject<HTMLDivElement>;
        const ref1 = searchInput.find(SuggestionInput).at(1).props().innerRef as RefObject<HTMLDivElement>;

        if (ref0 && ref1) {
            expect(ref0.current).toBe(document.activeElement);
            expect(ref1.current).not.toBe(document.activeElement);
        } else {
            expect(false).toBe(true);
        }
    })
})
