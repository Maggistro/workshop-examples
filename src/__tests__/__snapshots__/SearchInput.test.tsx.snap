// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for SearchInput should render the complete tree 1`] = `"<div><span>test</span><input><div tabindex=\\"0\\" class=\\"dummy\\"></div></div><div><span>test</span><input><div tabindex=\\"0\\" class=\\"dummy\\"></div></div>"`;
